src/main/kotlin/gg/skytils/skytilsmod/utils/MojangUtil.kt

```diff
@@ -65,6 +73,13 @@ object MojangUtil {
         }
     }
 
+    suspend fun queueUsernameForUUID(name: String): UUID? {
+        val username = name.lowercase()
+        usernameQueue.emit(username to System.currentTimeMillis())
+
+        return uuidFlow.first { it.first == username }.second
```

Comment:
if 2 queue the same username what will happen?

Reply:
It should still emit. Only `StateFlow` will disregard repeated consecutive elements.